cmd/fleet/cron.go

```diff
@@ -153,142 +218,47 @@ func cronVulnerabilities(
 			err := os.MkdirAll(vulnPath, 0o755)
 			if err != nil {
 				level.Error(logger).Log("databases-path", "creation failed, returning", "err", err)
-				return
+				sentry.CaptureException(err)
+				return err
 			}
 		}
 	}
 
-	ticker := time.NewTicker(10 * time.Second)
-	for {
-		level.Debug(logger).Log("waiting", "on ticker")
-		select {
-		case <-ticker.C:
-			level.Debug(logger).Log("waiting", "done")
-			ticker.Reset(config.Vulnerabilities.Periodicity)
-		case <-ctx.Done():
-			level.Debug(logger).Log("exit", "done with cron.")
-			return
-		}
-		if config.Vulnerabilities.CurrentInstanceChecks == "auto" {
-			if locked, err := ds.Lock(ctx, lockKeyVulnerabilities, identifier, 1*time.Hour); err != nil {
-				level.Error(logger).Log("msg", "Error acquiring lock", "err", err)
+	if !vulnDisabled {
+		level.Debug(logger).Log("vulnerabilities", "checking for recent vulnerabilities")
+		level.Debug(logger).Log("vuln-path", vulnPath)
+		recentVulns := checkVulnerabilities(ctx, ds, logger, vulnPath, config, appConfig.WebhookSettings.VulnerabilitiesWebhook)
+		if len(recentVulns) > 0 {
+			if err := webhooks.TriggerVulnerabilitiesWebhook(ctx, ds, kitlog.With(logger, "webhook", "vulnerabilities"),
+				recentVulns, appConfig, time.Now()); err != nil {
+
+				level.Error(logger).Log("err", "triggering vulnerabilities webhook", "details", err)
 				sentry.CaptureException(err)
-				continue
-			} else if !locked {
-				level.Debug(logger).Log("msg", "Not the leader. Skipping...")
-				continue
 			}
 		}
+	}
 
-		if !vulnDisabled {
```

Comment:
@lucasmrod, did we move this part somewhere else? It was a relatively recent change and I'm not seeing it elsewhere so I think this was lost in one of the merges.

Reply:
Great catch, missed this new code added recently.

Reply:
Re-added, please review :)